File: app/src/main/java/se/jakob/knarkklocka/WakeLocker.java
package se.jakob.knarkklocka;

import android.content.Context;
import android.os.PowerManager;

public class WakeLocker {
    private static final String TAG = "WakeLocker";
    private static PowerManager.WakeLock wakeLock;

    public static void acquire(Context context) {
        if (wakeLock != null) {
            PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
            if (pm != null) {
                wakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK | PowerManager.ACQUIRE_CAUSES_WAKEUP | PowerManager.ON_AFTER_RELEASE, TAG);
                wakeLock.acquire(30000);
            }
        }
    }

    public static void release() {
        if (wakeLock != null) {
            wakeLock.release();
            wakeLock = null;
        }
    }
}

File: app/src/main/java/se/jakob/knarkklocka/WakeLocker.kt
package se.jakob.knarkklocka

import android.content.Context
import android.os.PowerManager
import android.util.Log

object WakeLocker {
    private const val TAG = "WakeLocker"
    private var wakeLock: PowerManager.WakeLock? = null

    fun acquire(context: Context) {
        val pm = context.getSystemService(Context.POWER_SERVICE) as PowerManager
        wakeLock = wakeLock ?: pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK or PowerManager.ACQUIRE_CAUSES_WAKEUP or PowerManager.ON_AFTER_RELEASE, TAG)
        wakeLock?.acquire(3000)
        if (BuildConfig.DEBUG) {
            Log.d(TAG, "Acquired WakeLock")
        }
    }

    fun release() {
        wakeLock?.let {
            it.release()
            wakeLock = null
            if (BuildConfig.DEBUG) {
                Log.d(TAG, "Released WakeLock")
            }
        }
    }
}
